### PR TITLE
ENH: allow start=range_object in RangeIndex constructor

### LINKED ISSUE #12067: API: should the RangeIndex constructor accept start=range_index?
xref #11892 (see changes and discussion)

RangeIndex, a new Int64Index subclass that will be used as the new default index (currently) has the following signature: `RangeIndex(start=None, stop=None, step=None, name=None...)`

For converting an `xrange`/`range` object into a RangeIndex, we have the constructor `RangeIndex.from_range`.

Although it isn't documented, the current constructor supports accepting a `RangeIndex` instances as the `start` argument, in which case a copy of the original index is produced, e.g., `RangeIndex(RangeIndex(0, 10, 1)) -> RangeIndex(0, 10, 1)`.

I think this is a mistake: it is reusing the `start` parameter for a different type of argument, at odds with the documented behavior and predictable function signatures. If a user wants to create another RangeIndex from a RangeIndex, they can simply use the Index construtor, e.g., `Index(RangeIndex(...))`.

@jreback argues that not accepting a `RangeIndex` as the first argument to `RangeIndex` would be an API break, because it would create an Index constructor that is not idempotent. It would be "a major break with the current Index design".

Thoughts from anyone else?

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #12067
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

simplifies `Index.__new__` a bit

### CODE REVIEW

**Comment by jreback** [comment]
hah I originally implemented this but then we settled on `.from_range`. I would be +1 in using this. cc @jorisvandenbossche

File: pandas/core/indexes/range.py [hunk 1]
```diff
@@ -118,6 +118,8 @@ def __new__(
         if isinstance(start, RangeIndex):
             start = start._range
             return cls._simple_new(start, name=name)
```
Comment (topper-123) [inline pandas/core/indexes/range.py:120]: If the above two lines were replaced with `return start.copy(name=name)`, the `_cache`attribute would be reused on the new object and we'd get a performance improvement in cases where we'd have to re-access the cache.

Comment (jbrockmendel) [reply pandas/core/indexes/range.py:120]: neat, will update

**Comment by topper-123** [comment]
I'm +1 on this change. If we call something RangeIndex, it's reasonable that it takes a range in its constructor IMO.

**Comment by jorisvandenbossche** [comment]
(no strong opinion here)

**Comment by jreback** [comment]
lgtm. I think ok to also deprecated `.from_range`

**Comment by topper-123** [comment]
In the RangeIndex class doc stringm in the `Parameters` section, I think the line `start : int (default: 0), or other RangeIndex instance` should be changed to `start : int (default: 0), range or other RangeIndex instance`

**Comment by jbrockmendel** [comment]
> In the RangeIndex class doc stringm in the Parameters section, I think the line start : int (default: 0), or other RangeIndex instance should be changed to start : int (default: 0), range or other RangeIndex instance

good idea.  will add this to my upcoming core.indexes TYP branch if thats OK

**Pull request merged**